subprojects/configuration-cache/src/main/kotlin/org/gradle/configurationcache/ConfigurationCacheAwareBuildTreeWorkController.kt

```diff
@@ -47,6 +47,7 @@ class ConfigurationCacheAwareBuildTreeWorkController(
                 workExecutor.execute(result.graph)
             }
         }
+        @Suppress("SENSELESS_COMPARISON")
```

Comment:
💭  See https://youtrack.jetbrains.com/issue/KT-55542 for some more discussion on the problem.